dataset/data_modules/text_data_module.py

```diff
@@ -0,0 +1,47 @@
+from abc import abstractmethod
+from typing import Any, Callable
+
+import torch
+from torch.nn.utils.rnn import pad_sequence
+
+from dataset.base_data_module import BaseContrastiveDataModule
+from dataset.classification_datasets.text_dataset import TextDataset
+
+
+class TextDataModule(BaseContrastiveDataModule):
+    def __init__(
+            self,
+            dataset_name: str,
+            batch_size: int,
+            num_classes: int,
+            is_test: bool = False,
+            transform: Callable = None
+    ):
+        super().__init__(
+            dataset_name=dataset_name,
+            batch_size=batch_size,
+            is_test=is_test,
+            transform=transform,
+            num_classes=num_classes
+        )
+
+    @abstractmethod
```

Comment:
It is strange that you mark a method as abstract and then immediately implement it. Maybe it's better to remove this decorator?